keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -51,223 +48,175 @@
+The enhancement describes situations when such option can/cannot be used, why it's Kubernetes who must care about such a mount option, and possible breaking changes of the new Kubernetes behavior.
+
 ## Motivation
 
 ### SELinux intro
 On Linux machines with SELinux in enforcing mode, SELinux tries to prevent users that escaped from a container to access the host OS and also to access other containers running on the host.
 It does so by running each container with unique *SELinux context* (such as `system_u:system_r:container_t:s0:c309,c383`) and labeling all content on all volumes with the corresponding label (`system_u:object_r:container_file_t:s0:c309,c383`).
-Only process with the context `...:container_t:s0:c309,c383` can access files with label `container_file_t:s0:c309,c383`, even if the process runs as root.
-Therefore rogue user cannot access potentially secret data of other containers, because volumes of each container have different label.
+Only process with the context `...:container_t:s0:c309,c383` can access files with label `container_file_t:s0:c309,c383`.
+Therefore, a rogue user who escaped boundaries of its container cannot access data of other containers, because volumes of each container have different label.
+Even processes running as root (UID 0) are denied access to these files, unless they run with the right SELinux context. 
 
-In further text, we're going to shorten both `system_u:system_r:container_t:s0:c309,c383` (context of a process) and `system_u:object_r:container_file_t:s0:c309,c383` (label of a file) to `s0:309:383`.
+In the further text, we're going to shorten both `system_u:system_r:container_t:s0:c309,c383` (context of a process) and `system_u:object_r:container_file_t:s0:c309,c383` (label of a file) to `s0:309:383`.
+We also assume that the SELinux is enabled on the system. This KEP has absolutely no effect on systems that run without SELinux. Kubelet already knows that SELinux is not enabled and does not do anything with it. 
 
 See [SELinux documentation](https://selinuxproject.org/page/NB_MLS) for more details.
 
+
 ### SELinux context assignment
 In Kubernetes, the SELinux context of a pod is assigned in two ways:
 1. Either it is set by user in PodSpec or Container: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/.
 1. If not set in Pod/Container, the container runtime will allocate a new unique SELinux context and assign it to a pod (container) by itself.
 
-### Volumes
-Currently, Kubernetes *knows*(*) which volume plugins supports SELinux (i.e. supports extended attributes on a filesystem the plugin provides).
+### Volume mounting
+Linux kernel, with SELinux compiled in, allows `mount -o context=s0:c309,c383 <what> <where>` to mount a volume and pretend that all files on the volume have given SELinux label.
+
+Following conditions must be met:
+* It must be the first mount of the volume! It does not work for subsequent mounts of the volume, incl. bind-mounts.
```

Comment:
After a volume has been mounted with a selinux label, does the bind mount associated with safe subpath handling (to surface any subpaths from the volume to containers) work properly and inherit the selinux label in the original mount?

Reply:
Yes, bind mounts inherit `-o context` from the original mount.